Spigot-Server-Patches/0544-Add-Player-teleportRelative-and-Player-teleportWitho.patch

```diff
@@ -0,0 +1,51 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Anrza <andrzejrzeczycki314@gmail.com>
+Date: Wed, 15 Jul 2020 12:28:43 +0200
+Subject: [PATCH] Add Player#teleportRelative and
+ Player#teleportWithoutRotation
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+index b832ece4948323e60fabecdeee5c9051b65c5e8e..1839082cc5e56b201cec722ded7db83324dfcc4e 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+@@ -18,6 +18,7 @@ import java.net.InetSocketAddress;
+ import java.net.SocketAddress;
+ import java.util.ArrayList;
+ import java.util.Collection;
++import java.util.EnumSet;
+ import java.util.HashMap;
+ import java.util.HashSet;
+ import java.util.LinkedHashMap;
+@@ -60,6 +61,7 @@ import net.minecraft.server.PacketPlayOutMap;
+ import net.minecraft.server.PacketPlayOutNamedSoundEffect;
+ import net.minecraft.server.PacketPlayOutPlayerInfo;
+ import net.minecraft.server.PacketPlayOutPlayerListHeaderFooter;
++import net.minecraft.server.PacketPlayOutPosition;
+ import net.minecraft.server.PacketPlayOutSpawnPosition;
+ import net.minecraft.server.PacketPlayOutStopSound;
+ import net.minecraft.server.PacketPlayOutTitle;
+@@ -2064,6 +2066,23 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
+         }
+         throw new RuntimeException("Unknown settings type");
+     }
++
++    @Override
++    public void teleportWithoutRotation(Location location) {
++        Set<PacketPlayOutPosition.EnumPlayerTeleportFlags> set = EnumSet.noneOf(PacketPlayOutPosition.EnumPlayerTeleportFlags.class);
++        set.add(PacketPlayOutPosition.EnumPlayerTeleportFlags.X_ROT);
++        set.add(PacketPlayOutPosition.EnumPlayerTeleportFlags.Y_ROT);
++        //getLocation().getYaw() returns the player's head rotation , but playerConnection.internalTeleport compares with the body rotation
++        getHandle().playerConnection.a(location.getX(), location.getY(), location.getZ(), getHandle().yaw, location.getPitch(), set);
```

Comment:
Needs obfhelper

Reply:
Thanks for the tip! Do you mean a method like PlayerConnection#teleport(...) {a(...)}?

Reply:
We put a method "above" the obfuscated method that names it, search for "OBFHELPER" and youll find an example